File: mockspresso-api/src/main/java/com/episode6/hackit/mockspresso/api/MockspressoPlugin.java
package com.episode6.hackit.mockspresso.api;

import com.episode6.hackit.mockspresso.Mockspresso;
import org.jetbrains.annotations.NotNull;

/**
 * An interface that can encapsulate multiple related mockspresso components into a single plugin.
 */
public interface MockspressoPlugin {
  @NotNull Mockspresso.Builder apply(@NotNull Mockspresso.Builder builder);
}

File: mockspresso-api/src/main/java/com/episode6/hackit/mockspresso/api/MockspressoPlugin.kt
package com.episode6.hackit.mockspresso.api

import com.episode6.hackit.mockspresso.Mockspresso

/**
 * An interface that can encapsulate multiple related mockspresso components into a single plugin.
 */
interface MockspressoPlugin {
  fun apply(builder: Mockspresso.Builder): Mockspresso.Builder
}